### PR TITLE
MULE-8512: Allow default processing strategy to be configured per app

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: core/src/main/java/org/mule/construct/AbstractPipeline.java [hunk 1]
```diff
@@ -91,6 +90,24 @@ protected MessageProcessor createPipeline() throws MuleException
         return builder.build();
     }
 
+    protected ProcessingStrategy createDefaultProcessingStrategy()
```
Comment (dfeist) [inline core/src/main/java/org/mule/construct/AbstractPipeline.java]: Default can't only go in MuleConfiguration?  Or is this because of need to test without MueContext etc.?

File: modules/spring-config/src/main/resources/META-INF/mule.xsd [hunk 2]
```diff
@@ -1366,7 +1366,7 @@
                         </xsd:documentation>
                     </xsd:annotation>
                 </xsd:attribute>
-                <xsd:attribute name="flowEndingWithOneWayEndpointReturnsNull" type="substitutableBoolean" default="false">
+                <xsd:attribute name="flowEndingWithOneWayEndpointReturnsNull" type="substitutableBoolean" use="optional">
```
Comment (dfeist) [inline modules/spring-config/src/main/resources/META-INF/mule.xsd]: Is this change related to this PR?

File: modules/spring-config/src/main/resources/META-INF/mule.xsd [hunk 3]
```diff
@@ -2474,74 +2543,13 @@
                         </xsd:documentation>
                     </xsd:annotation>
                 </xsd:attribute>
-                 <xsd:attribute name="processingStrategy" use="optional" default="default">
```
Comment (dfeist) [inline modules/spring-config/src/main/resources/META-INF/mule.xsd]: Add simple comment, saying where default is defined now.

File: modules/spring-config/src/main/resources/META-INF/mule.xsd (hunk 2 above)
Comment (marianogonzalez) [inline modules/spring-config/src/main/resources/META-INF/mule.xsd]: no, this looks like a mistake of some sort

File: core/src/main/java/org/mule/construct/AbstractPipeline.java (hunk 1 above)
Comment (marianogonzalez) [inline core/src/main/java/org/mule/construct/AbstractPipeline.java]: this method is a fallback for creating the processing strategy you'll use if the user hasn't either set something on the flow, nor the configuration, nor the system property. Will add javadoc to better explain
